bugwarrior/services/todoist.py

```diff
@@ -0,0 +1,220 @@
+                all_users.append(user)
+        return all_users
+
+    def get_issues(self):
+        tasks_iter = self._api.filter_tasks(query=self.filter)
+        return tasks_iter
+
+
+class TodoistIssue(Issue):
+    ASSIGNEE = "todoistassignee"
+    ASSIGNER = "todoistassigner"
+    CONTENT = "todoistcontent"
+    DESCRIPTION = "todoistdescription"
+    DURATION = "todoistduration"
+    ID = "todoistid"
+    SECTION = "todoistsection"
+    URL = "todoisturl"
+
+    PRIORITY_MAP = {
+        4: "H",
+        3: "M",
+        2: "L",
+        1: None,
+    }
+
+    UDAS = {
+        ID: {
+            "type": "string",
+            "label": "Todoist ID",
+        },
+        CONTENT: {
+            "type": "string",
+            "label": "Todoist Content",
+        },
+        DESCRIPTION: {
+            "type": "string",
+            "label": "Todoist Description",
+        },
+        DURATION: {
+            "type": "string",
+            "label": "Todoist Duration",
+        },
+        SECTION: {
+            "type": "string",
+            "label": "Todoist Section",
+        },
+        ASSIGNEE: {
+            "type": "string",
+            "label": "Todoist Assignee",
+        },
+        ASSIGNER: {
+            "type": "string",
+            "label": "Todoist Assigner",
+        },
+        URL: {
+            "type": "string",
+            "label": "Todoist URL",
+        },
+    }
+
+    UNIQUE_KEY = (ID, ID)
+
+    # replace characters that cause escaping issues in teh description like [] and "
+    # this is a workaround for https://github.com/ralphbean/taskw/issues/172
+    def _unescape_content(self, content):
+        return (
+            content.replace('"', "'")  # prevent &dquote; in task details
+            .replace("[", self.config.char_open_bracket)  # prevent &open; and &close;
+            .replace("]", self.config.char_close_bracket)
+        )
+
+    def to_taskwarrior(self):
+        default_time = time(0, 0, 0)
+        # use due date "scheduled".
+        # adjust timezone to use local time for "floating" dates
+        if self.record.due and type(self.record.due.date) is datetime:
+            if self.record.due.timezone:
+                scheduled = self.record.due.date
+            else:
+                scheduled = self.record.due.date.replace(tzinfo=None)
+        else:
+            scheduled = (
+                datetime.combine(self.record.due.date, default_time, tzinfo=None)
+                if self.record.due
+                else None
+            )
+
+        # use deadline as "due".
+        # deadline if set is only a date with no time or timezone. adjust to locla time
+        due = (
+            datetime.combine(self.record.deadline.date, default_time, tzinfo=None)
+            if self.record.deadline
+            else None
+        )
+
+        task = {
+            "project": self.extra["project"],
+            "priority": self.PRIORITY_MAP[self.record.priority],
```

Comment:
I'm realizing we ought to make this change in the logseq service too because this will respect the `default_priority` option.
```suggestion
            "priority": self.get_priority(),
```